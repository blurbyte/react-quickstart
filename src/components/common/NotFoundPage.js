import React from 'react';
import {IndexLink} from 'react-router';

const NotFoundPage = () => (
  <section>
    <h1>404 Not Found</h1>
    <IndexLink to="/">Take me back to Main Page</IndexLink>
  </section>
);

export default NotFoundPage;
