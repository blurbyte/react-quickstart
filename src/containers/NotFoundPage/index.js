import React from 'react';
import Helmet from 'react-helmet';

// components
import ContentWrapper from 'components/ContentWrapper';
import MainHeadline from 'components/MainHeadline';
import NavLink from 'components/NavLink';

class NotFoundPage extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <ContentWrapper>
        <Helmet>
          <title>404 Not Found</title>
        </Helmet>
        <MainHeadline>404 Not Found</MainHeadline>
        <NavLink to="/">Take me back to Main Page</NavLink>
      </ContentWrapper>
    );
  }
}

export default NotFoundPage;
