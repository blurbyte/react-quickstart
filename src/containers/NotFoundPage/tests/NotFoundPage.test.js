import React from 'react';
import renderer from 'react-test-renderer';

import NotFoundPage from '../index';

// mocked components
jest.mock('react-helmet', () => 'Helmet');
jest.mock('components/ContentWrapper', () => 'ContentWrapper');
jest.mock('components/MainHeadline', () => 'MainHeadline');
jest.mock('components/NavLink', () => 'NavLink');

describe('<NotFoundPage />', () => {
  it('should render correctly', () => {
    // component init
    const component = renderer.create(
      <NotFoundPage />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
