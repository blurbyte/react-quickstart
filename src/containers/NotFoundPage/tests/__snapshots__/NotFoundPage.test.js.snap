// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should render correctly 1`] = `
<ContentWrapper>
  <Helmet>
    <title>
      404 Not Found
    </title>
  </Helmet>
  <MainHeadline>
    404 Not Found
  </MainHeadline>
  <NavLink
    to="/"
  >
    Take me back to Main Page
  </NavLink>
</ContentWrapper>
`;
